homeassistant/components/fibaro/sensor.py

```diff
@@ -113,7 +113,15 @@ async def async_setup_entry(
         # main sensors are created even if the entity type is not known
         entities.append(FibaroSensor(device, entity_description))
 
-    for platform in (Platform.COVER, Platform.LIGHT, Platform.SENSOR, Platform.SWITCH):
+    for platform in (
```

Comment:
As these are all supported platforms except `SENSOR`, which is handled a few lines above, I would suggest combining the code and looping over all items in `hass.data[DOMAIN][entry.entry_id][FIBARO_DEVICES]`.

This has the advantage that if a new platform is supported in the feature, the additional sensors will also be added (if supported) without an extra code change.

Reply:
Good hint, but you swapped scene and sensor. Scenes are as well in this list and are quite different and therefore should not be looked up (but yes in a future reafctoring that should be changed).

So your suggestion will not work without a bigger refactoring.

Reply:
@edenhaus Plese check my comment and add a feedback if you agree so we can continue with this PR. Many thanks.